### PR TITLE
bugfix: `start_process` ensuring passing of `TMPDIR` when needed

### PR DESCRIPTION
**Pull request by roi-granulate**
## Description
https://github.com/Granulate/gprofiler/pull/921#discussion_r1743316860

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [ ] I have added tests for new logic.

### CODE REVIEW

File: gprofiler/utils/__init__.py [hunk 1]
```diff
@@ -148,7 +148,9 @@ def start_process(
         else:
             # explicitly remove our directory from LD_LIBRARY_PATH
             env = env if env is not None else os.environ.copy()
```
Comment (roi-granulate) [inline gprofiler/utils/__init__.py:150]:
```suggestion
            env = env or os.environ.copy()
```

Comment (Jongy) [reply gprofiler/utils/__init__.py:150]: No, prefer explicit `is None`

File: gprofiler/utils/__init__.py [hunk 2]
```diff
@@ -148,7 +148,9 @@ def start_process(
...
-            env.update({"LD_LIBRARY_PATH": ""})
+            if "TMPDIR" not in env and "TMPDIR" in os.environ:
+                env["TMPDIR"] = os.environ.get("TMPDIR")
```
Comment (Jongy) [inline gprofiler/utils/__init__.py:152]:
```suggestion
                env["TMPDIR"] = os.environ["TMPDIR"]`
```
makes it clear that the value can't be `None`.

**Review by Jongy — CHANGES_REQUESTED** [review]
.

File: gprofiler/utils/__init__.py [hunk 3]
```diff
@@ -148,7 +148,9 @@ def start_process(
         else:
             # explicitly remove our directory from LD_LIBRARY_PATH
```
Comment (Jongy) [inline gprofiler/utils/__init__.py:149]: Move the comment to its relevant code line now. And add another comment explaining the TMPDIR ("respect the TMPDIR we were run with in all cases")

**Review by Jongy — DISMISSED** [review]

**Review by Jongy — APPROVED** [review]

**Pull request merged**